source/features/restore-file.tsx

```diff
@@ -49,11 +44,23 @@ async function commitFileContent(menuItem: Element, content: string, filePath: s
 	let {pathname} = menuItem.previousElementSibling as HTMLAnchorElement;
 	// Check if file was deleted by PR
 	if (menuItem.closest('[data-file-deleted="true"]')) {
-		menuItem.textContent = 'Undeleting…';
+		void showToast(async () => {
+			// Fake a task
+		}, {
+			message: 'Undeleting…',
+			doneMessage: 'Undeleted!',
+			onDone: ToastOnDoneState.success,
+		});
 		const [nameWithOwner, headBranch] = select('.head-ref')!.title.split(':');
 		pathname = `/${nameWithOwner}/new/${headBranch}?filename=${filePath}`;
 	} else {
-		menuItem.textContent = 'Committing…';
+		void showToast(async () => {
+			// Fake a task
+		}, {
+			message: 'Committing changes…',
+			doneMessage: 'Changes committed!',
+			onDone: ToastOnDoneState.success,
+		});
```

Comment:
This new interface and the fake task defeats the purpose of passing a task to showToast. Refer to my implementation suggestion to see how I'd expect this to work.

The problem is that `showTask` currently doesn't support show progress, so we can pass a callback function to the task to allow doing that.

For example:

```ts
async function restoreFile(progress: (message: string) => void) {
	await api.v4(…);
	progress('Committing…')
	await pushForm();
}
```

and the regular:

```ts
await showToast(restoreFile, {
	message: 'Restoring…',
	doneMessage: 'Restored'
})

Reply:
Also any progress message should keep the loading gif, not change it to a "done" state, which should only appear when the task is done.

Reply:
Thanks!